core/src/main/scala/zquery/ZQuery.scala

```diff
@@ -199,6 +201,24 @@ sealed trait ZQuery[-R, +E, +A] { self =>
       a     <- runCache(cache)
     } yield (cache, a)
 
+  /**
+   * Summarizes a query by computing some value before and after execution,
+   * and then combining the values to produce a summary, together with the
+   * result of execution.
+   */
+  final def summarized[R1 <: R, E1 >: E, B, C](f: (B, B) => C)(summary: ZIO[R1, E1, B]): ZQuery[R1, E1, (C, A)] =
+    for {
+      start <- ZQuery.fromEffect(summary)
+      value <- self
+      end   <- ZQuery.fromEffect(summary)
+    } yield (f(start, end), value)
+
+  /**
+   * Returns a new query that executes this one and times the execution.
+   */
+  final def timed: ZQuery[R with Clock, E, (Duration, A)] =
```

Comment:
Nice! Do you mind calling it in `ApolloTracing.apolloTracingField`?

Reply:
Will do!